osprey_flask_app/routes.py

```diff
@@ -43,18 +47,79 @@ def osprey_route():
     Returns output netCDF file after Convolution process.
     """
     args = request.args
-    arg_dict = create_full_arg_dict(args)
-    outpath = run_full_rvic(arg_dict)
     try:
-        outpath_url = requests.get(outpath)
+        arg_dict = create_full_arg_dict(args)
+        datetime.datetime.strptime(arg_dict["run_startdate"], "%Y-%m-%d-%H")
+        datetime.datetime.strptime(arg_dict["stop_date"], "%Y-%m-%d")
+        file_list = ["pour_points", "uh_box", "routing", "domain", "input_forcings"]
+        for f in file_list:
+            if "fileServer" in arg_dict[f]:
+                http_response = requests.head(arg_dict[f])
+                if http_response.status_code != 200:
+                    return Response(
+                        f"File not found on THREDDS using http: {arg_dict[f]}",
+                        status=400,
+                    )
+            elif "dodsC" not in arg_dict[f]:
+                open(arg_dict[f], "r")
+            else:
+                try:
+                    netCDF4.Dataset(
+                        arg_dict[f] + "?lon[0:1]"
+                    )  # Load tiny slice of dataset
+                except OSError:
+                    return Response(
+                        f"File not found on THREDDS using OPeNDAP: {arg_dict[f]}",
+                        status=400,
+                    )
+
+        rvic_thread = threading.Thread(
+            target=lambda q, arg: q.put(run_full_rvic(arg)), args=(que, arg_dict)
+        )
```

Comment:
I have the queue because I believe that `threading` doesn't preserve return values after joining back to the main thread, so the queue stores that return value to be picked up later in the `output` url. Not sure if there's a cleaner way to do this.